src/AzureAppConfigurationImpl.ts

```diff
@@ -55,17 +92,23 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
             };
             if (this.requestTracingEnabled) {
                 listOptions.requestOptions = {
-                    customHeaders: this.customHeaders()
+                    customHeaders: this.customHeaders(requestType)
                 }
             }
 
             const settings = this.client.listConfigurationSettings(listOptions);
 
             for await (const setting of settings) {
                 if (setting.key) {
-                    const [key, value] = await this.processAdapters(setting);
-                    const trimmedKey = this.keyWithPrefixesTrimmed(key);
-                    keyValues.push([trimmedKey, value]);
+                    const keyValuePair = await this.processKeyValues(setting);
+                    keyValues.push(keyValuePair);
+                }
+                // update etag of sentinels if refresh is enabled
+                if (this.refreshEnabled) {
+                    const matchedSentinel = this.sentinels.find(s => s.key === setting.key && s.label === setting.label);
+                    if (matchedSentinel) {
+                        matchedSentinel.etag = setting.etag;

```

Comment:
Just confirming if this update to `matchedSentinel` will also update the corresponding etag in `this.sentinels`?

Reply:
Yes, it will. `find` returns the reference of the object. By modifying its properties, it affects this.sentinels which actually is a list of references.